### DISCUSSION TITLE
Custom input on toolbar

### DISCUSSION THREAD

**Discussion by cubecleveland** [geoman-io/leaflet-geoman#1066] [Questions & Help] [answered]
Hello,

Is theer a way to put a custom Input on the toolbar ? Like a text box r is it limited to only buttons?

**Comment by Falke-Design** [comment]
You can add elements with default DOM functions but not over Geoman

**Comment by cubecleveland** [comment, marked as answer]
Thnk you and is there any way to takje the toolbar outside the leaflet-control-container into an outside element?

**Reply by Falke-Design** [reply]
The most functions that are used in the Toolbar are public so you could read about it in the docs and then you can create a custom Toolbar and adding the same icons by using the same CSS classes as in the Geoman-Toolbar.

Like:
```
<button id='draw-circle'>Draw Circle</button>
```
```js
var btn = document.getElementById('draw-circle');
L.DomEvent.on(btn, 'click', (e)=>{
   map.pm.enableDraw('Circle');
});
```
